go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -519,12 +519,8 @@ func (nb *northBoundClient) deletePodGWRoute(routeInfo *ExternalRouteInfo, podIP
 	}
 
 	node := util.GetWorkerFromGatewayRouter(gr)
-	// The gw is deleted from the routes cache after this func is called, length 1
-	// means it is the last gw for the pod and the hybrid route policy should be deleted.
-	if entry := routeInfo.PodExternalRoutes[podIP]; len(entry) == 1 {
-		if err := nb.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
-			return fmt.Errorf("unable to delete hybrid route policy for pod %s: err: %v", routeInfo.PodName, err)
-		}
+	if err := nb.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
+		return fmt.Errorf("unable to delete hybrid route policy for pod %s: err: %v", routeInfo.PodName, err)
```

Comment:
I don't really understand what `entry := routeInfo.PodExternalRoutes[podIP]; len(entry) == 1` meant or why we can do without it, even after reading the explanation on the commit message, will have to trust you here

Reply:
this is a mistake. I was confusing exgw pod IP with the served pod IP. We still need this check here, although I will change it to be <= 1, just in case somehow we get here with no gateways, we should still try to remove the IP from the policy/addr set.

Reply:
Yeah, definitely I don't get it. Here we are just removing one gw for the pod but apparently there can be more than one and in such case we should not remove the pod IP from the AS which is no related to a specific gw

Reply:
I wrote my thing before you wrote yours, all clear.